core/server/master/src/main/java/alluxio/master/file/replication/ReplicationChecker.java

```diff
@@ -167,12 +178,18 @@ private void check(Set<Long> inodes, ReplicationHandler handler, Mode mode) {
                 maxReplicas = file.getReplicationDurable();
               }
               if (currentReplicas > maxReplicas) {
-                requests.add(new ImmutableTriple<>(inodePath.getUri(), blockId,
+                requests.add(new ReplicationRequest(Mode.EVICT, inodePath.getUri(), blockId,
                     currentReplicas - maxReplicas));
               }
               break;
             case REPLICATE:
               int minReplicas = file.getReplicationMin();
+              // check if it is correctly replicated
+              if (currentReplicas > 0 && (!checkCorrectlyReplicated(file, blockInfo))) {
+                // some copy is in memory and it is not correctly replicated
+                requests.add(new ReplicationRequest(Mode.EVICT, inodePath.getUri(), blockId, 1));
+                currentReplicas--;
+              }
```

Comment:
We plan to add the logic for moving the incorrectly place copy in a later PR?

Reply:
i had this change in another branch.. really not sure how this change got into this PR.